packages/realm/src/Realm.ts

```diff
@@ -1483,12 +1547,78 @@ export declare namespace Realm {
+      export type ChangeEventId = ChangeEventIdType;
+      /** @deprecated Please use named imports */
+      export type CountOptions = CountOptionsType;
+      /** @deprecated Please use named imports */
+      export type DeleteEvent<T extends Document> = DeleteEventType<T>;
+      /** @deprecated Please use named imports */
+      export type DeleteResult = DeleteResultType;
+      /** @deprecated Please use named imports */
+      export type Document<IdType = any> = DocumentType<IdType>;
+      /** @deprecated Please use named imports */
+      export type DocumentKey<IdType> = DocumentKeyType<IdType>;
+      /** @deprecated Please use named imports */
+      export type DocumentNamespace = DocumentNamespaceType;
+      /** @deprecated Please use named imports */
+      export type DropDatabaseEvent = DropDatabaseEventType;
+      /** @deprecated Please use named imports */
+      export type DropEvent = DropEventType;
+      /** @deprecated Please use named imports */
+      export type Filter = FilterType;
+      /** @deprecated Please use named imports */
+      export type FindOneAndModifyOptions = FindOneAndModifyOptionsType;
+      /** @deprecated Please use named imports */
+      export type FindOneOptions = FindOneOptionsType;
+      /** @deprecated Please use named imports */
+      export type FindOptions = FindOptionsType;
+      /** @deprecated Please use named imports */
+      export type InsertEvent<T extends Document> = InsertEventType<T>;
+      /** @deprecated Please use named imports */
+      export type InsertManyResult<IdType> = InsertManyResultType<IdType>;
+      /** @deprecated Please use named imports */
+      export type InsertOneResult<IdType> = InsertOneResultType<IdType>;
+      /** @deprecated Please use named imports */
+      export type InvalidateEvent = InvalidateEventType;
+      /** @deprecated Please use named imports */
+      export type MongoDBCollection<T extends Document> = MongoDBCollectionType<T>;
+      /** @deprecated Please use named imports */
+      export type NewDocument<T extends Document> = NewDocumentType<T>;
+      /** @deprecated Please use named imports */
+      export type OperationType = OperationTypeType;
+      /** @deprecated Please use named imports */
+      export type RenameEvent = RenameEventType;
+      /** @deprecated Please use named imports */
+      export type ReplaceEvent<T extends Document> = ReplaceEventType<T>;
+      /** @deprecated Please use named imports */
+      export type Update = UpdateType;
+      /** @deprecated Please use named imports */
+      export type UpdateDescription = UpdateDescriptionType;
+      /** @deprecated Please use named imports */
+      export type UpdateEvent<T extends Document> = UpdateEventType<T>;
+      /** @deprecated Please use named imports */
+      export type UpdateOptions = UpdateOptionsType;
+      /** @deprecated Please use named imports */
+      export type UpdateResult<IdType> = UpdateResultType<IdType>;
```

Comment:
These might their type argument defaults too.

Reply:
Are you saying we should introduce defaults to all type arguments? (I used what the "old" types were using)

Reply:
No sorry. I mis remembered for some reason 👍